.github/workflows/increment-version-on-merge.yml

```diff
@@ -38,13 +38,21 @@ jobs:
         with:
           ref: main
           fetch-depth: 0
-          persist-credentials: false
```

Comment:
If this is a big no no, then using Octokit might be a better option.   But having this here causes this [error](https://github.com/bradyclifford/git-version-lite/actions/runs/4327307781/jobs/7555785531).

Reply:
This should be fine.

Reply:
I'm not sure why the arg needs to be removed.  Is this because the create version and major tags step has been added?  This was specifically added because we're running on ubuntu latest and didn't necessarily want any creds leftover.

Reply:
I added details in the PR description about this.

> `persist-credentials: false` on the `actions/checkout` must be removed or we get this [error](https://github.com/bradyclifford/git-version-lite/actions/runs/4327307781/jobs/7555785531)

Reply:
Yes, it's removed so that this workflow can later use the git CLI commands.

The build.yml file is running on the same runners as this workflow and doesn't have the `persist-credentials: false` setting. The only thing it really does is store the `GITHUB_TOKEN` in the working directory's git config.

There are three failsafes that prevent this authentication from leaking
- The checkout action has a [built-in `post` function](https://github.com/actions/checkout/blob/main/src/main.ts#L31) that runs at the end of the job and cleans up by [removing authentication](https://github.com/actions/checkout/blob/main/src/git-source-provider.ts#L287): 
- The runner is wiped at the end of each job
- The `GITHUB_TOKEN` is invalidated at the end of each job run